app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb

```diff
@@ -182,12 +182,13 @@ def vm_and_template_taggings(resource, tags_inventory_objects)
   def flavors
     collector.flavors.each do |flavor|
       memory = flavor&.dig(:memory, :value)
-      memory_mb = Integer(memory) * 1024 if memory
+      disk = flavor[:disks].first&.dig(:size, :value) || 0
```

Comment:
Is there any guarantee that the first disk is the root disk?

Reply:
`.first` here is due to the way IBM Cloud has structured their response. `flavor[:disks]` is an array of hashes however, there is only ever a single hash in this array. Flavors that use multiple disks as returned by the API, only use disks that are of the same size and as a result when there are multiple disks it is specified in the response like this:
```
"disks"=>[{
  "quantity"=>{"type"=>"fixed", "value"=>2},
  "size"=>{"type"=>"fixed", "value"=>2640},
  "supported_interface_types"=>{"default"=>"virtio_blk", "type"=>"enum", "values"=>["virtio_blk"]}
  }]
```
I couldn't find a way to determine which disk is the root but `:root_disk_size` would be set accordingly since the disks are of the same size

Reply:
Ah okay so even if there are multiple disks there's still only one array entry, got it thanks